src/script/components/asset/controls/MediaButton.test.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Wire
+ * Copyright (C) 2019 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {AssetTransferState} from 'src/script/assets/AssetTransferState';
+import TestPage from 'Util/test/TestPage';
+import MediaButton, {MediaButtonProps} from 'Components/asset/controls/MediaButton';
+import {FileAsset} from 'src/script/entity/message/FileAsset';
+
+class MediaButtonTestPage extends TestPage<MediaButtonProps> {
+  constructor(props?: MediaButtonProps) {
+    super(MediaButton, props);
+  }
+
+  getPlayButton = () => this.get('[data-uie-name="do-play-media"]');
+  getPauseButton = () => this.get('[data-uie-name="do-pause-media"]');
+  getAssetLoader = () => this.get('[data-uie-name="status-loading-media"]');
+  clickOnPlayButton = () => this.click(this.getPlayButton());
+  clickOnPauseButton = () => this.click(this.getPauseButton());
+}
+
+describe('MediaButton', () => {
+  const videoElement = document.createElement('video');
```

Comment:
Probably fine but I would prefer to not share the state between tests. Imagine the test execution order changes and the video is still playing while another test expects it to be stopped.

As a rule of thumb I would propose:
- Prefer functions that create test data multiple times over test values that are used multiple times.

Reply:
Ok, I will just create a new element per test.

Reply:
Here it is: https://github.com/wireapp/wire-webapp/pull/10483/commits/ffd27251ffb76c19f71d60a797a01ee473ea186f